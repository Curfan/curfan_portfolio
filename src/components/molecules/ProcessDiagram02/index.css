.process-diagram02 {
	display: flex;
	position: relative;
	justify-content: space-around;

	& > .step {
		display: flex;
		flex: 1;
		flex-direction: column;
		justify-content: center;
		align-items: center;
		cursor: pointer;

		& > .marker {
			background-color: #1e222b;
		}

		& > .label {
			margin-top: 25px;
			color: rgba(255, 255, 255, 0.12);
			transition: color 0.3s ease-in;
		}

		&.active {
			& > .label {
				color: white;
			}
		}
	}

	& > .process-line {
		position: absolute;
		top: 12px;
		left: 0;
		width: 100%;
		height: 10px;
		background-repeat: repeat-x;
		z-index: -1;
	}
}
