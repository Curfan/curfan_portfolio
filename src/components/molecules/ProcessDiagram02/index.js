import React from 'react';
import classnames from 'classnames';

import Marker from 'components/atoms/Marker';

import ProcessLine from 'images/process-line.png';

import styles from './index.css';

const ProcessDiagram02 = ({ className, activeIndex = 0, process = [], onClickItem = () => {} }) => (
	<div className={classnames(styles.processDiagram02, className)}>
		{process.map((step, index) => (
			<div
				key={step}
				className={classnames(styles.step, { [styles.active]: activeIndex === index })}
				role="button"
				tabIndex="0"
				onClick={() => onClickItem(index)}
				onKeyPress={() => {}}
			>
				<Marker className={styles.marker} active={activeIndex === index} />
				<div className={styles.label}>{step}</div>
			</div>
		))}

		<div className={styles.processLine} style={{ backgroundImage: `url(${ProcessLine})` }} />
	</div>
);

export default ProcessDiagram02;
