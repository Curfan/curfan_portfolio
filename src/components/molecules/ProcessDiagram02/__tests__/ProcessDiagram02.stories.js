import React from 'react';
import { storiesOf } from '@storybook/react';

import ProcessDiagram02 from 'components/molecules/ProcessDiagram02';

const stories = storiesOf('molecules/ProcessDiagram02', module);

stories.add('__interactive', () => (
	<ProcessDiagram02 process={['Synergies', 'Neobards', 'Trend Mirco', 'Dcard', 'T T S']} />
));
