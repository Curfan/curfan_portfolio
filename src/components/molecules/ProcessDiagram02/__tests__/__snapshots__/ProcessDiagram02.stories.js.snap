// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    class="processDiagram02"
  >
    <div
      class="step active"
      role="button"
      tabindex="0"
    >
      <div
        class="marker marker active"
      >
        <span />
      </div>
      <div
        class="label"
      >
        Synergies
      </div>
    </div>
    <div
      class="step"
      role="button"
      tabindex="0"
    >
      <div
        class="marker marker"
      >
        <span />
      </div>
      <div
        class="label"
      >
        Neobards
      </div>
    </div>
    <div
      class="step"
      role="button"
      tabindex="0"
    >
      <div
        class="marker marker"
      >
        <span />
      </div>
      <div
        class="label"
      >
        Trend Mirco
      </div>
    </div>
    <div
      class="step"
      role="button"
      tabindex="0"
    >
      <div
        class="marker marker"
      >
        <span />
      </div>
      <div
        class="label"
      >
        Dcard
      </div>
    </div>
    <div
      class="step"
      role="button"
      tabindex="0"
    >
      <div
        class="marker marker"
      >
        <span />
      </div>
      <div
        class="label"
      >
        T T S
      </div>
    </div>
    <div
      class="processLine"
      style="background-image: url(test-file-stub);"
    />
  </div>
</DocumentFragment>
`;
